numba/cuda/compiler.py

```diff
@@ -801,7 +806,14 @@ def _prepare_args(self, ty, val, stream, retr, kernelargs):
             parent = ctypes.c_void_p(0)
             nitems = c_intp(devary.size)
             itemsize = c_intp(devary.dtype.itemsize)
-            data = ctypes.c_void_p(driver.device_pointer(devary))
+
+            ptr = driver.device_pointer(devary)
+
+            if config.CUDA_USE_CUDA_PYTHON:
+                ptr = int(ptr)
+
+            data = ctypes.c_void_p(ptr)
+
```

Comment:
I can't see "quite a bit" of this - and in this particular case, I wouldn't be able to use a function that does `void(int(pointer))` because the `int` part is only applied when using the Nvidia binding - am I misunderstanding this suggestion?

Reply:
I probably made that comment after seeing two or three instances, looking further, it  seems restricted to a few places, fine to leave I think.